pkg/cmd/server/server.go

```diff
@@ -832,3 +826,39 @@ func (s *server) runProfiler() {
 		s.logger.WithError(errors.WithStack(err)).Error("error running profiler http server")
 	}
 }
+
+// runLiveness starts an HTTP server to handle liveness checks for the velero server.
+// Note that liveness checks should have generous timeouts, since Velero will query
+// object storage, which is possibly a network call across the internet.
+func (s *server) runLiveness() {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/livez", func(w http.ResponseWriter, r *http.Request) {
+		// Refresh the discovery helper, since it may be stale if the velero API group was
+		// created after the pod.
+		s.discoveryHelper.Refresh()
+		if err := s.veleroResourcesExist(); err != nil {
+			s.logger.Warn(err)
+			http.NotFound(w, r)
+			return
+		}
+
+		if err := s.validateBackupStorageLocations(); err != nil {
```

Comment:
Current logic of validateBackupStorageLocations would return error if there is any invalid BackupStorageLocation.  Should Velero continue function if there is at least 1 valid BackupStorageLocation?  If so, validateBackupStorageLocations should be enhanced to return more than just an error.  For example return number of valid BackupStorageLocation and at this point we decide Velero should continue work if some valid BackupStorageLocation present.

Reply:
This validation will change after https://github.com/vmware-tanzu/velero/pull/2617.

We will change the validation logic to only error if there are no valid BackupStorageLocations. Else it will be only warnings if there's any invalid BSL.